Source/WebCore/html/track/WebVTTElement.h

```diff
@@ -66,28 +69,89 @@ class WebVTTElement final : public Element {
         static NeverDestroyed<QualifiedName> voiceAttr(nullAtom(), "voice"_s, nullAtom());
         return voiceAttr;
     }
-    
+
     static const QualifiedName& langAttributeName()
     {
         static NeverDestroyed<QualifiedName> voiceAttr(nullAtom(), "lang"_s, nullAtom());
         return voiceAttr;
     }
 
-private:
-    WebVTTElement(WebVTTNodeType, Document&);
-
-    bool isWebVTTElement() const override { return true; }
+protected:
+    WebVTTElementImpl(WebVTTNodeType nodeType, AtomString language)
+        : m_isPastNode { false }
+        , m_webVTTNodeType { static_cast<unsigned>(nodeType) }
+        , m_language { language }
+    {
+    }
+    virtual ~WebVTTElementImpl() = default;
+    virtual Element& toElement() = 0;
 
     unsigned m_isPastNode : 1;
     unsigned m_webVTTNodeType : 4;
-    
+
     AtomString m_language;
 };
 
+class WebVTTElement final : public WebVTTElementImpl, public Element {
+    WTF_MAKE_ISO_ALLOCATED(WebVTTElement);
+public:
+    Ref<Element> cloneElementWithoutAttributesAndChildren(Document& document) override { return WebVTTElementImpl::cloneElementWithoutAttributesAndChildren(document); }
```

Comment:
final

Reply:
Does override replace final or used in conjunction with it?

Reply:
It replaces it; `final` means the method cannot be overridden, and `override` means it can

Reply:
I think it's used in conjunction; `override` means the function is overriding a function in its parent, and `final` means any subclasses of this class can't override it.

Reply:
We spoke in person- Eric means that final should replace override